src/peer/db/statement-db.js

```diff
@@ -0,0 +1,168 @@
+// @flow
+
+const path = require('path')
+const Knex = require('knex')
+const pb = require('../../protobuf')
+
+import type { StatementMsg, SimpleStatementMsg, EnvelopeStatementMsg, CompoundStatementMsg } from '../../protobuf/types'
+
+const MIGRATIONS_DIR = path.join(__dirname, 'migrations')
+
+type StatementDBOptions = {
+  filename: string,
+}
+
+const DefaultOptions: StatementDBOptions = {
+  filename: ':memory:'
+}
+
+class StatementDB {
+  _db: Object
+  _migrated: boolean
+
+  constructor (options: StatementDBOptions = DefaultOptions) {
+    this._db = Knex({
+      client: 'sqlite3',
+      connection: {
+        filename: options.filename
+      },
+      useNullAsDefault: true
+    })
+    this._migrated = false
+  }
+
+  /**
+   * Accessor for the underlying sql DB (via knex query builder).
+   * Use this instead of directly accessing the `_db` instance,
+   * to ensure that the db migrations are run before accessing
+   * the db.
+   * @returns {*}
+   */
+  sqlDB (): Promise<Object> {
+    if (this._migrated) return Promise.resolve(this._db)
+
+    return this._db.migrate.latest({directory: MIGRATIONS_DIR})
+      .then(() => {
+        this._migrated = true
```

Comment:
Yeah, I was concerned something like that might happen. I'm kind of surprised knex doesn't have an internal way to handle this.